wolfssl/src/ssl.rs

```diff
@@ -922,7 +890,10 @@ impl<IOCB: IOCallbacks> Session<IOCB> {
         // owned by the `Session`. See `register_io_context` below for
         // an argument as to why IO will be stopped (by releasing
         // `WOLFSSL`) before that box is dropped.
-        let io = unsafe { &*(ctx as *mut IOCB) };
+        //
+        // Since `io` is owned by the `Session` and we know that `try_read` takes the lock.
+        // XXX needs thought since lock doesn't protect io
```

Comment:
I think this case is fine, since now `Session::try_read` take mutable Session when in turn calls `wolfSSl_read` -> `io_recv_shim`
So at this point, io_recv_shim will have ctx aka. Session.ssl will have exclusive access and hence safe to access.

Please let me know if my understanding is wrong. If not, please update the comment to reflect the same.

Reply:
I think you are correct. IIRC I wrote this before I'd worked out the whole `&mut self` change.

Reply:
Updated the comment (and in the write path too)